File: hydrograph.engine/hydrograph.engine.cascading/src/main/java/hydrograph/engine/flow/utils/ExecutionTrackingListener.java
package hydrograph.engine.flow.utils;

import java.util.List;

import cascading.stats.StatsListener;
import hydrograph.engine.cascading.integration.RuntimeContext;
import hydrograph.engine.execution.tracking.ComponentInfo;

public interface ExecutionTrackingListener extends StatsListener {

	public void addListener(RuntimeContext runtimeContext);
	
	public List<ComponentInfo> getStatus();
	
}

File: hydrograph.engine/hydrograph.engine.spark/src/main/scala/hydrograph/engine/spark/executiontracking/plugin/ExecutionTrackingListener.scala
package hydrograph.engine.spark.executiontracking.plugin

import hydrograph.engine.execution.tracking.ComponentInfo
import hydrograph.engine.spark.flow.RuntimeContext
import org.apache.spark.scheduler.SparkListener
import org.apache.spark.sql.SparkSession

import scala.collection.mutable
import scala.collection.mutable.ListBuffer

/**
  * Created by gurdits on 12/15/2016.
  */
trait ExecutionTrackingListener extends SparkListener{

  def addListener(runtimeContext: RuntimeContext): Unit ={
//    sparkSession.sparkContext.addSparkListener(this)
  }

  def getStatus():mutable.HashSet[ComponentInfo]

}
